src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -70,13 +70,17 @@ public function populate($values)
         unset($loadedVars['']);
 
         foreach ($values as $name => $value) {
-            if (!isset($loadedVars[$name]) && (isset($_ENV[$name]) || isset($_SERVER[$name]) || false !== getenv($name))) {
+            $notHttpName = 0 !== strpos($name, 'HTTP_');
+            // don't check existence with getenv() because of thread safety issues
+            if (!isset($loadedVars[$name]) && (isset($_ENV[$name]) || (isset($_SERVER[$name]) && $notHttpName))) {
                 continue;
             }
 
             putenv("$name=$value");
             $_ENV[$name] = $value;
-            $_SERVER[$name] = $value;
+            if ($notHttpName) {
```

Comment:
with this, env vars that start with `HTTP_*` are do not populate `$_SERVER`.
this is to mitigate security issues with $_SERVER mixing several sources (env+http headers+etc).